code/script.js

```diff
@@ -4,16 +4,127 @@
 // Your code goes here
 alert(
   `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
-)
+);
+
+let Name = prompt("What is your name?");
+if (Name !== null) {
+  alert("Hi, " + Name + "! Welcome!");
+} else {
+  alert("Hi there! Nice to meet you!");
+}
 
 // Step 2 - Food choice
-// Your code goes here
+let userInput = prompt("Please select a food type:\n1. Pizza\n2. Pasta\n3. Salad");
 
-// Step 3 - Subtype choice
-// Your code goes here
+let selectedOption = parseInt(userInput);
+
+// Declare the following variables outside the if-else blocks
+let pizzaSubtype, pastaSubtype, saladSubtype;
+
+if (selectedOption === 1) {
+  alert("You have selected Pizza!");
+
+  // Step 3 - Subtype choice
+  pizzaSubtype = prompt("Please select a pizza subtype:\n1. Margherita\n2. Pepperoni\n3. Veggie");
+
+  if (pizzaSubtype === "1") {
+    alert("You have selected Margherita Pizza!");
+  } else if (pizzaSubtype === "2") {
+    alert("You have selected Pepperoni Pizza!");
+  } else if (pizzaSubtype === "3") {
+    alert("You have selected Veggie Pizza!");
+  } else {
+    alert("Invalid choice. Please select a valid subtype.");
+  }
+
+} else if (selectedOption === 2) {
+  alert("You have selected Pasta!");
+
+  // Step 3 - Subtype choice
+  pastaSubtype = prompt("Please select a pasta:\n1. Spaghetti\n2. Fettuccine Alfredo\n3. Pasta Agli Oglio");
+
+  if (pastaSubtype === "1") {
+    alert("You have selected Spaghetti Pasta!");
+  } else if (pastaSubtype === "2") {
+    alert("You have selected Fettuccine Alfredo Pasta!");
+  } else if (pastaSubtype === "3") {
+    alert("You have selected Pasta Agli Oglio!");
+  } else {
+    alert("Invalid choice. Please select a valid subtype.");
+  }
+
+} else if (selectedOption === 3) {
+  alert("You have selected Salad!");
+
+  // Step 3 - Subtype choice
+  saladSubtype = prompt("Please select a salad:\n1. Caesar Salad\n2. Greek Salad\n3. Garden Salad");
+
+  if (saladSubtype === "1") {
+    alert("You have selected Caesar Salad!");
+  } else if (saladSubtype === "2") {
+    alert("You have selected Greek Salad!");
+  } else if (saladSubtype === "3") {
+    alert("You have selected Garden Salad!");
+  } else {
+    alert("Invalid choice. Please select a valid subtype.");
+  }
+} else {
+  alert("Invalid choice. Please select a valid option.");
+}
```

Comment:
if you used switch break and array that would be better and made your code shorter but nice job